polynomial/__init__.py

```diff
@@ -96,30 +101,39 @@ def calculate(self, x):
 
     def __getattr__(self, name):
         """Get coefficient by letter name: ax^n + bx^{n-1} + ... + yx + z."""
-        if len(name) == 1 and name in string.ascii_uppercase:
-            return self.__getattr__(name.lower())
-        if len(name) == 1 and name in string.ascii_lowercase:
-            return self[self.degree - (ord(name) - ord('a'))]
-
-        return object.__getattr__(self, name)
+        if len(name) != 1:
+            return object.__getattribute__(self, name)
+        elif name in string.ascii_letters:
+            return self[self.degree - ord(name.lower()) + ord('a')]
+        else:
+            raise AttributeError("attribute {0} is not defined for Polynomial.".format(name))
 
     def __setattr__(self, name, new_value):
         """Set coefficient by letter name: ax^n + bx^{n-1} + ... + yx + z."""
-        if len(name) == 1 and name in string.ascii_uppercase:
-            self.__setattr__(name.lower(), new_value)
-        elif len(name) == 1 and name in string.ascii_lowercase:
-            self[self.degree - (ord(name) - ord('a'))] = new_value
-        else:
+        if len(name) != 1:
             object.__setattr__(self, name, new_value)
+        elif name in string.ascii_letters:
+            self[self.degree - ord(name.lower()) + ord('a')] = new_value
+        else:
+            raise AttributeError("attribute {0} is not defined for Polynomial.".format(name))
 
     def __getitem__(self, degree):
         """Get the coefficient of the term with the given degree."""
+        if isinstance(degree, slice):
+            indices = degree.indices(self.degree)
+            return [self._vector[degree] for degree in range(*indices)]
+
         if degree > self.degree or degree < 0:
             raise IndexError("Attempt to get coefficient of term with \
 degree {0} of a {1}-degree polynomial".format(degree, self.degree))
         return self._vector[degree]
 
     def __setitem__(self, degree, new_value):
+        if isinstance(degree, slice):
+            indices = degree.indices(self.degree)
+            for degree, value in zip(range(*indices), new_value):
+                self._vector[degree] = value
+
         """Set the coefficient of the term with the given degree."""
```

Comment:
The docstring should be the first line of the method, there should be no code above it in the scope.